### ISSUE TITLE
boto3-stubs error: "resource" is not a known member of module

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#891]
Setup to reproduce the behavior:
```
pip install 'boto3-stubs[s3]==1.14.30'
pip install 'boto3==1.14.30'
python -m mypy_boto3
```

Run pyright on the following code sample:

```python
import boto3

s3_resource = boto3.resource('s3', region_name='us-east-1')
```

pyright output:
```
No configuration file found.
stubPath /private/tmp/t/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
/private/tmp/t/test.py
  3:21 - error: "resource" is not a known member of module (reportGeneralTypeIssues)
1 error, 0 warnings
Completed in 0.576sec
```

**Expected output**

No type errors. 

NB: When only `boto3` is installed (ie: without stubs) there are no type errors.

pyright 1.1.58

**Comment by erictraut** [comment]
The boto3 stub file (__init__.pyi) is attempting to import symbols from an implementation file (boto3_init.py). This is really sketchy, and it's not something that pyright supports unless you tell it that you want it to extract types from library code.

I consider this a bug in the boto3-stubs. I recommend filing a bug against that repo or file a PR so these stubs avoid importing a non-stub file. Stub files should also not use try blocks, since they are a runtime construct, and pyi files are never executed.

While you are waiting for a fix for boto3-stubs, you can enable Pyright's "python.analysis.useLibraryCodeForTypes" setting or use `--lib` on the command line. This is not a perfect workaround because type inferred from library code are invariably incomplete. That's why it's important to use stub files.

**Comment by tekumara** [comment]
Thanks for analysis @erictraut. `--lib` does solve this particular error but introduces other issues in the project for the libraries that don't have stubs (eg: numpy). And it doesn't look like there is an ability to selectively enable `--lib` for some files, or dependencies, and not others.
